pkg/jobrunaggregator/jobtableprimer/job_name_generator.go

```diff
@@ -0,0 +1,263 @@
+package jobtableprimer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+)
+
+type jobNameGenerator struct {
+	periodicURLs      []string
+	releaseConfigURLs []string
+	releases          []jobrunaggregatorapi.ReleaseRow
+}
+
+var (
+	periodicURLTemplates = []string{
+		"https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/release/openshift-release-release-%s-periodics.yaml",
+		"https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-%s-periodics.yaml",
+	}
+	releaseConfigURLTemplates = []string{
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s-arm64.json",
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s-ci.json",
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s-multi.json",
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s-ppc64le.json",
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s-s390x.json",
+		"https://raw.githubusercontent.com/openshift/release/master/core-services/release-controller/_releases/release-ocp-%s.json",
+	}
+)
+
+func newJobNameGenerator() *jobNameGenerator {
+	generator := &jobNameGenerator{
+		periodicURLs: []string{
+			"https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml",
+			"https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/multiarch/openshift-multiarch-master-periodics.yaml",
+
+			"https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/release/openshift-release-release-4.10-periodics.yaml",
```

Comment:
nit - do we even need this any longer or can we just use the templates for all the defined releases?